src/translate.ts

```diff
@@ -30,32 +30,30 @@ const localeToPluralization = {
   fr: frenchPluralization,
 };
 
-interface TranslateOptions {
-  locale: string;
-  translations: Translations;
-  globalTranslations?: Translations;
-  convertMissingKey?: ConvertMissingKey;
-  transforms?: Transforms;
-  preprocess?: Preprocess;
-  postprocess?: Postprocess;
-}
-
-export default function translate({
+export default function translate<T extends Translations>({
   locale,
   translations,
   globalTranslations = {},
   convertMissingKey,
   transforms = {},
   preprocess,
   postprocess,
-}: TranslateOptions): TranslateLocal {
+}: {
+  locale: string;
+  translations: T;
+  globalTranslations?: Translations;
+  convertMissingKey?: ConvertMissingKey;
+  transforms?: Transforms;
+  preprocess?: Preprocess;
+  postprocess?: Postprocess;
+}): TranslateLocal<T> {
   function usePhrase(trs: Translations, key: string) {
     return trs.hasOwnProperty(key) && trs[key].hasOwnProperty(locale)
       ? trs[key][locale]
       : (convertMissingKey && convertMissingKey(key, trs)) || key;
   }
   function usingTranslations(trs: Translations) {
-    function tr(key: string, interpolation?: Interpolation) {
+    function tr(key: string & keyof T, interpolation?: Interpolation) {
```

Comment:
by this you want to allow passing any string event if it's not defined in dictionary?

Reply:
this is in fact correct - the intersection type of these two - so `keyof T`s that are also strings

![image](https://user-images.githubusercontent.com/614768/147224378-3cf259c5-b682-4159-8c9c-37a87764160b.png)

Reply:
there is a test that requires a `// @ts-expect-error missing key` for missing keys so this is covered

Reply:
![image](https://user-images.githubusercontent.com/614768/147224595-cb6dd436-0ae2-46d0-9d4f-fa05407fdfdf.png)

Reply:
interesting I didn't know that you can narrow the type of object index properties to be only strings by using `string & keyof T`